### PR TITLE
Add new configure option --with-mixedrefs for mixed pointer mode

### PR DESCRIPTION
**Pull request by sharon-wang**
JDK8 port of https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/359.

Related Issue: https://github.com/eclipse/openj9/issues/8878

### CODE REVIEW

**Comment by sharon-wang** [comment]
FYI @keithc-ca @gacholio

File: jdk/make/closed/autoconf/generated-configure.sh [hunk 1]
```diff
@@ -4546,7 +4550,7 @@ VS_SDK_PLATFORM_NAME_2017=
 
 
 # Do not change or remove the following line, it is needed for consistency checks:
-DATE_WHEN_GENERATED=1604507962
+DATE_WHEN_GENERATED=1605209646
```
Comment (keithc-ca) [inline jdk/make/closed/autoconf/generated-configure.sh:4553]: Please include corresponding changes to `jdk/make/closed/autoconf/generated-configure.sh`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by sharon-wang** [comment]
The `generated-configure.sh` file automatically removed `runstatedir`-related code. Is this supposed to happen?

**Comment by keithc-ca** [comment]
> The `generated-configure.sh` file automatically removed `runstatedir`-related code. Is this supposed to happen?

That's a sign that you're using a slightly older version of autoconf (even though it might report it's version 2.69). The one in Ubuntu 18.04 or newer won't do that.

**Comment by sharon-wang** [comment]
Yes, I was building on OSX and it was reporting v2.69 for autoconf. I've run the autogen script again on Ubuntu 20.04 -- `runstatedir`-related code is now un-removed.

**Comment by keithc-ca** [comment]
Jenkins compile aix,osx,win,win32,zlinux jdk8

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**